src/arithmetic/arithmetic_expression.c

```diff
@@ -298,8 +298,17 @@ static AR_EXP_Result _AR_EXP_EvaluateFunctionCall(AR_ExpNode *node, const Record
 	return res;
 }
 
-static inline void _AR_EXP_UpdateEntityIdx(AR_OperandNode *node, const Record r) {
-	node->variadic.entity_alias_idx = Record_GetEntryIdx(r, node->variadic.entity_alias);
+static inline AR_EXP_Result _AR_EXP_UpdateEntityIdx(AR_OperandNode *node, const Record r) {
+	int entry_alias_idx = Record_GetEntryIdx(r, node->variadic.entity_alias);
+	if(entry_alias_idx == INVALID_INDEX) {
+		char *error;
+		asprintf(&error, "There is no such node or edge named %s", node->variadic.entity_alias);
```

Comment:
A bit of a missleading message, entity_alias isn't necessarily a node or an edge,
think about projections.